Recorder xml

gilserrancoli opened the pull request:
Those are the changes to enable the reading of .osim files in the AD version of OpenSim. The main changes consist of enabling readFromXML... functions. 
I added a test to read arm26_noMuscles .osim and gait2354_simbody osim file with no muscles and no Yamaguchi knee functions (gait2354_simbody_noMuscles_nokneeYamag.osim). 
Pendulum_2dof_osim External function has also been created to read the model directly from the .osim file and prepared to run an OCP.

gilserrancoli committed:
most changes to OpenSim to enable osim reading (and also most Contact Geometries)

gilserrancoli committed:
more changes to enable osim reading and add test to read .osim files

gilserrancoli committed:
more changes to enable the reading of osim files

gilserrancoli committed:
add .osim folders as shared files to run the tests

gilserrancoli committed:
organize files for tests

gilserrancoli committed:
remove system pause from test

gilserrancoli committed:
add Pendulum2dof.osim in the shared folder of models

antoinefalisse commented on OpenSim/External_Functions/Pendulum_2dofs_osim/CMakeLists.txt:
```diff
@@ -0,0 +1,6 @@
+target_link_libraries(${TEST_TARGET} osimSimulation)
+set_target_properties(${TEST_TARGET} PROPERTIES
+    FOLDER "External_Functions")
+OpenSimCopySharedTestFiles(Pendulum2dof.osim)
```
Can you add an empty line in the end?

antoinefalisse commented on OpenSim/External_Functions/Pendulum_2dofs_osim/Pendulum_2dofs_osim.cpp:
```diff
@@ -0,0 +1,147 @@
+/*  This code describes the OpenSim model and the skeleton dynamics
```
Can you adjust the tabs to make it look nice? Thanks.

antoinefalisse commented on OpenSim/Simulation/Model/ElasticFoundationForce.cpp:
```diff
@@ -1,385 +1,385 @@
-//
-//} // end of namespace OpenSim
+/* -------------------------------------------------------------------------- *
+ *                    OpenSim:  ElasticFoundationForce.cpp                    *
```
Do we support that with AD? I don't think so...I don't think we should uncomment if we do not support...

antoinefalisse commented on OpenSim/Simulation/SimbodyEngine/Body.cpp:
```diff
@@ -25,6 +25,7 @@
 // INCLUDES
 //=============================================================================
 #include "Body.h"
+#include <OpenSim/Common/XMLDocument.h>
```
Strange that we need to add those

antoinefalisse commented on OpenSim/Simulation/SimbodyEngine/CustomJoint.cpp:
```diff
@@ -30,7 +30,7 @@
 #include <OpenSim/Common/LinearFunction.h>
 #include "simbody/internal/MobilizedBody_FunctionBased.h"
-
+#include <OpenSim/Common/XMLDocument.h>
```
Idem

antoinefalisse commented on OpenSim/Simulation/SimbodyEngine/Joint.cpp:
```diff
@@ -31,6 +31,7 @@
 #include <OpenSim/Common/ScaleSet.h>
 #include "simbody/internal/Constraint.h"
 #include "simbody/internal/MobilizedBody_Ground.h"
+#include <OpenSim/Common/XMLDocument.h>
```
Idem

antoinefalisse commented on OpenSim/Tests/CMakeLists.txt:
```diff
@@ -19,5 +19,6 @@ if(BUILD_TESTING)
     add_subdirectory(ExampleLuxoMuscle)
     add_subdirectory(AnalysisPluginExample)
     add_subdirectory(BodyDragExample)
+    add_subdirectory(testReadOsimFile)
```
I would include Recorder or AD as part of the test name

antoinefalisse commented on OpenSim/Tests/shared/arm26_noMuscles.osim:
```diff
@@ -0,0 +1,1045 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Where does this model come from?

antoinefalisse commented on OpenSim/Tests/testReadOsimFile/CMakeLists.txt:
```diff
@@ -0,0 +1,53 @@
+
```
Can you clean this? There are some unnecessary comments.

antoinefalisse commented on OpenSim/Tests/testReadOsimFile/testReadOsimFile.cpp:
```diff
@@ -0,0 +1,100 @@
+/* -------------------------------------------------------------------------- *
```
Let's work on the test in the end. I think we should test more features.

antoinefalisse commented on cgeneration/Pendulum_2dofs_osim/CMakeLists.txt:
```diff
@@ -0,0 +1,16 @@
+	ARCHIVE DESTINATION lib
+	RUNTIME DESTINATION bin
+	)
+		
```
Seems to be an empty line in the end

antoinefalisse reviewed (changes requested):
@gilserrancoli, I had a first look. I wonder whether it is necessary to uncomment everything. In the end we do not support certain things with AD, such as the elastic foundation model. So I am not sure we should make that available.

gilserrancoli committed:
add external functions for smooth mesh-based contact model and models from Movement Simulation subject

gilserrancoli committed:
update signs of rotations when calculating forces

gilserrancoli committed:
Update TrackSim_2_kneeCont.cpp

gilserrancoli committed:
Apply changes to the contact model

Avoid issue with order of tibial faces (columns in pairs) and change pressure calculation

gilserrancoli committed:
Update Visualization_Forces_movement to be consistent with the dll

Still need to work on it

gilserrancoli committed:
Changes for debug and in _kneeCont.cpp to take into account last face

gilserrancoli committed:
Changes in TrackSim debug and non debug for clarification and k=1e4

gilserrancoli committed:
add mesh-based contact model with tangential force, and other external function of Movement Simulation Subject

gilserrancoli committed:
Update TrackSim_2_kneeCont.cpp

gilserrancoli committed:
update cpp files knee contact

gilserrancoli committed:
add external functions and modify Track2Sim_2_kneeCont to use any knee geometry

gilserrancoli committed:
Update TrackSim_2_kneeCont_forDebug

gilserrancoli committed:
Adapt the contact model to work with cylinders to compute multipliers

gilserrancoli committed:
Fix bug in TrackSim_2_kneeCont and adapt contact model for TrackSim_2_kneeCont_debug

gilserrancoli committed:
Fix bug in TrackSim_2_kneeCont

gilserrancoli committed:
Fixed bugs in TackSim_2_kneeCont_forDebug

gilserrancoli committed:
update k in CheckContact to 1e3

gilserrancoli committed:
Update after correcting the bug in potential contact pairs

gilserrancoli committed:
Update with RighRatHindlimb_Zhong.cpp and Sphere_mesh.cpp

gilserrancoli committed:
update sphere cpp

gilserrancoli committed:
Update Sphere_mesh.cpp

gilserrancoli committed:
Update Sphere_mesh.cpp

gilserrancoli committed:
Update mesh with HuntCrossley contact

gilserrancoli committed:
Update RightRatHindlimb_Zhong.cpp

RatHidLimb version data May

gilserrancoli committed:
Update rathindlimb with inertia properties (but not consistent with experiments)

gilserrancoli committed:
update mass and inertia properties with mass from experiments

gilserrancoli committed:
Knee prosthesis for the workshop 2024 in Barcelona

gilserrancoli committed:
Update new dll for subject Mov Sim and update TrackSim to work with several PCs

gilserrancoli committed:
add PredSim external function adapted for the Mesh based contact model

gilserrancoli committed:
changes rat model for May2025 model (before using 2D angles)

gilserrancoli committed:
add TrackSim_2_kneeCont_KCFasinput

gilserrancoli committed:
add cuntions to compute skeletal dynamics without KCF as input

gilserrancoli committed:
Bug corrected, include application of knee contact forces

gilserrancoli committed:
Create Rugby_generic_1.cpp

gilserrancoli committed:
Update files
